tests/lava/proc/io/test_injector.py

```diff
@@ -0,0 +1,537 @@
+        recv = Recv(shape=data_shape, buffer_size=num_steps)
+
+        injector.out_port.connect(recv.in_port)
+
+        run_condition = RunSteps(num_steps=num_steps)
+        run_cfg = Loihi2SimCfg()
+
+        send_data = np.random.random(size=(num_send, ) + data_shape)
+
+        def thread_2_fn() -> None:
+            for send_data_single_item in send_data:
+                injector.send(send_data_single_item)
+
+        thread_2 = threading.Thread(target=thread_2_fn,
+                                    daemon=True)
+        thread_2.start()
+
+        injector.run(condition=run_condition, run_cfg=run_cfg)
+
+        recv_var_data = recv.var.get()
+
+        injector.stop()
+
+        np.testing.assert_equal(recv_var_data, send_data)
+
+    def test_run_steps_non_blocking(self):
+        """Test that running the a Lava network involving the Injector
+        Process, with RunSteps(blocking=False), for multiple time steps,
+        with the main thread calling send, runs and terminates."""
+        np.random.seed(0)
+
+        data_shape = (1,)
+        buffer_size = 10
+
+        num_steps = 50
+        num_send = num_steps
+
+        injector = Injector(shape=data_shape, buffer_size=buffer_size)
+        recv = Recv(shape=data_shape, buffer_size=num_steps)
+
+        injector.out_port.connect(recv.in_port)
+
+        run_condition = RunSteps(num_steps=num_steps, blocking=False)
+        run_cfg = Loihi2SimCfg()
+
+        send_data = np.random.random(size=(num_send,) + data_shape)
+
+        injector.run(condition=run_condition, run_cfg=run_cfg)
+
+        for send_data_single_item in send_data:
+            injector.send(send_data_single_item)
+
+        injector.wait()
+
+        recv_var_data = recv.var.get()
+
+        injector.stop()
+
+        np.testing.assert_equal(recv_var_data, send_data)
+
+    def test_run_continuous(self):
+        """Test that running the a Lava network involving the Injector
+        Process, with RunContinuous(), for multiple time steps, for multiple
+        time steps, runs and terminates."""
+        np.random.seed(0)
+
+        data_shape = (1,)
+        buffer_size = 10
+
+        num_send = 50
+
+        injector = Injector(shape=data_shape, buffer_size=buffer_size)
+        recv = Recv(shape=data_shape, buffer_size=num_send)
+
+        injector.out_port.connect(recv.in_port)
+
+        run_condition = RunContinuous()
+        run_cfg = Loihi2SimCfg()
+
+        send_data = np.random.random(size=(num_send,) + data_shape)
+
+        injector.run(condition=run_condition, run_cfg=run_cfg)
+
+        for send_data_single_item in send_data:
+            injector.send(send_data_single_item)
+
+        injector.pause()
+        injector.wait()
+
+        recv_var_data = recv.var.get()
+
+        injector.stop()
+
+        np.testing.assert_equal(recv_var_data[:num_send // 10],
+                                send_data[:num_send // 10])
```

Comment:
The `:num_send // 10` needs to be changed to compare all sent vs received data or it needs to be documented.

Reply:
In the Injector, I added a comment explaining why this was happening.
In the Extractor, I got rid of it because it wasn't actually needed (i.e it now compares to all sent data).